crates/typst-library/src/foundations/symbol.rs

```diff
@@ -198,13 +198,22 @@ impl Symbol {
         #[variadic]
         variants: Vec<Spanned<SymbolVariant>>,
     ) -> SourceResult<Symbol> {
-        let mut list = Vec::new();
+        let mut list = Vec::<(EcoString, _)>::new();
         if variants.is_empty() {
             bail!(span, "expected at least one variant");
         }
         for Spanned { v, span } in variants {
-            if list.iter().any(|(prev, _)| &v.0 == prev) {
-                bail!(span, "duplicate variant");
+            if let Some((prev, _)) =
+                list.iter().find(|(prev, _)| same_modifiers(&v.0, prev))
```

Comment:
A bit unfortunate to have quadratic complexity on the find here. Together with the complexity of `same_modifiers` we are on $O(n^2 m^2)$ where $n$ is the number of variants and $m$ the number of modifiers.

Reply:
An alternative is to sort the modifiers in `list`, so that we can use binary search. But this will require additional allocations.

Reply:
In the happy case (i.e., no duplicate variant or 128-bit hash conflict), each variant now takes $O(m \log m)$ to process (where $m$ is its number of modifiers), for a total complexity of $O(n m \log m)$ where $n$ is the number of variants and $m$ the number of modifiers per variant. This comes at the cost of an allocation at each iteration of the loop.